### DISCUSSION TITLE
How to use streams with MySql (works for postgres)

### DISCUSSION THREAD

**Discussion by mpiorowski** [weiznich/diesel_async#86] [Q&A] [answered]
So i had an sql query, one as vec, one as stream. The vectors one works great, the stream don't, but the excatly same functions works when i change it to postgres (don't mind the linting errors in the methods, my lsp broke ;p)
```
pub async fn get_notes_by_user_uuid(
    mut conn: Object<AsyncMysqlConnection>,
    user_uuid: Vec<u8>,
) -> Result<Vec<DieselNote>, Status> {
    let note = notes
        .select(DieselNote::as_select())
        .filter(user_id.eq(user_uuid))
        .filter(deleted.is_null())
        .load(&mut conn)
        .await
        .map_err(|e| Status::internal(e.to_string()))?;
    Ok(note)
}

pub async fn get_notes_by_user_uuid_stream_stream(
    mut conn: Object<AsyncMysqlConnection>,
    user_uuid: Vec<u8>,
) -> Result<impl Stream<Item = QueryResult<DieselNote>>, Status> {
    let note = notes
        .select(DieselNote::as_select())
        .filter(user_id.eq(user_uuid))
        .filter(deleted.is_null())
        .load_stream(&mut conn)
        .await
        .map_err(|e| Status::internal(e.to_string()))?;
    Ok(note)
}
```
![image](https://github.com/weiznich/diesel_async/assets/26543876/6ef4f758-a3e4-41c3-a1e2-cef325c4c790)

**Comment by weiznich** [comment, marked as answer]
The issue here is that both stream types are not supposed to be equal. The postgres stream type is independent from the underlying connection, while the mysql stream type requires a connection for the whole time. That means the connection **must** outlive the stream. That's not the case in your example, as the connection is dropped by the end of the `get_notes_by_user_uuid_stream_stream` function, while the stream itself is returned by that function.
